### ISSUE TITLE
BE Spike to develop a standalone app review gathering script

### ISSUE THREAD

**Issue by dumathane** [department-of-veterans-affairs/va-mobile-app#6904]
## Description 
As a @lstraghalis, I want to review app store reviews for common themes and feedback, so that we can use that information in our development strategies.

As part of this story we want to:   

- Create tickets necessary to automate this process

## Acceptance Criteria
  

-  [ ] Create tickets necessary to automate this process

**Accessibility Requirements**
- [ ] @lstraghalis accessible.

## Notes & Open Questions

- Dependencies/Roadblocks:
    - Any internal/external dependencies?
    - Test accounts needed?
- Does this require QA?
    
- Dev Notes: 

**Here are my manual steps for both android and ios review compilation:**

iOS
----
1: https://itunes.apple.com/us/rss/customerreviews/id=1559609596/sortBy=mostRecent/json and copy the page and see if you need to include more. I had to go to https://itunes.apple.com/us/rss/customerreviews/page=2/id=1559609596/sortby=mostrecent/json?urlDesc=/customerreviews/id=1559609596/sortBy=mostRecent/json to get a few more entries.
2: Cutout feed entry and attribute lines from the list so you just have lines that start {"author":.
3: Macro to delete up to name.
4: find "}, "label":""}, "updated":{"label":" and replace with ~
5: find "}, "im:rating":{"label":" and replace with ~
6: find "}, "im:version":{"label":" and replace with ~
7: find "}, "id":{"label":" and replace with ~
8: find "}, "title":{"label":" and replace with ~
9: find "}, "content":{"label":" and replace with ~
10: macro to delete past ", "attributes".
11: Save the file and import into a google spreadsheet ~ delineated.
12: Delete column E (The unique ID's)
13: Add a header row with columns: User, Date/Time, Rating, Version, Title, Message.

Android
-------
1: Go to the Google Play Console page and select the DSVA Login
2: Click on the reviews tab under Download Reports.
3: In the top right select VA: Health and Benefits.
4: Download the specific month.
5: Import into new sheet and delete columns A, B, D, F, G, I, M, N, O, P
6: Sort by column F and delete all entries without text.
7: Sort by column C.
8: Move column A to the right of column D.

## Ticket Checklist

- [ ] Acceptance criteria defined
- [ ] Labels added (front-end, back-end, feature)
- [ ] Linked to an Epic

**Comment by jperk51** [comment]
My brief investigation revealed there is an API for each iOS and Android that supposedly should allow the fetching of app reviews. That is probably easier than doing a web crawler or something to follow the first couple steps @dumathane listed for each. Maybe it's not, but worth looking into for sure.

**Comment by Tonksthebear** [comment]
Apple App Store:
- As was already pointed out, we can get JSON responses fairly easily from the App Store. It doesn't seem we can specify date ranges, so we would basically copy the logic from the reddit scraper (sort by new, keep scrolling back and save the results of each new one we find). It would be fairly easy to create a ruby gem that does just this. There is also a [python script](https://github.com/facundoolano/app-store-scraper) similar to the Google Play Store one

Google Play Store:
- It's a bit more complicated than the App Store as there isn't an easy to read URL to hit in order to get a response that doesn't depend on Javascript (web scraping with javascript is VERY heavy). There is a python [library](https://github.com/JoMingyu/google-play-scraper) that found a url we can hit in order to get a response. We could either add Python as a dependency, or convert the python script to ruby so that it works without extra dependencies in our existing rails project.
 
TLDR: There are no ruby gems for either, but there are python scripts. Python would add more overhead to the maintenance and setup of the project. We could initially try a converter tool to turn the python into ruby and see what the outcome is. In either case, best case scenario would be making ruby gems (open source if possible), next would be adding python as a dependency to the project and using that. We should be able to scrape this easily though

**Comment by jperk51** [comment]
@Tonksthebear Reopened. Please make the necessary tickets before closing

**Comment by Tonksthebear** [comment]
App Store ticket: https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7042
Google Play Store ticket: https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7043
